File: SCRIPTS/1.1/abstracao_tv_com_get_set.js
class Tv {
    constructor () {
        this._relacaoCanal = Array(2,4,6,1)
        this._canalAtivo = 10
        this._volume = 5
    }
    //getters setters
    get canalAtivo() {
        return this._canalAtivo
    }

    set canalAtivo(canal) {
        if (this._relacaoCanal.indexOf(canal) !== -1) {
       this._canalAtivo = canal
        }
    }
}

let tv = new Tv()
console.log(tv.canalAtivo)
tv.canalAtivo = 4
console.log(tv.canalAtivo)

File: SCRIPTS/1.1/abstracao_tv_com_get_set.py
class Tv:
    def __init__(self):
        self._relacaoCanal = [2, 4, 6, 1]
        self._canalAtivo = 10
        self._volume = 5

    @property
    def canalAtivo(self):
        return self._canalAtivo

    @canalAtivo.setter
    def canalAtivo(self, newCanal):
        self._canalAtivo = newanal

tv = Tv()
tv.canalAtivo = 100
print(tv.canalAtivo)
